tests/cstdio/test_remove.cpp

```diff
@@ -52,7 +52,7 @@ void test() {
   err = std::remove("dir");
   assert(!err);
 
-  std::cout << "success";
+  std::cout << "success" << std::endl;
```

Comment:
Hmm, what made this test change necessary? Since we are making stdio more unbuffered than before, this feels like if something, this change would be going to the opposite direction?

Reply:
Previously all writes to std::cout were adding a newline automatically - I was sure there was a bug for that, but I can't seem to find it now?  Anyway the changes in this PR fix that so that you have to explicitly output a newline.

I could have changed the test output, but decided to change to code in this case - I can do the former instead if that's clearer?

Reply:
Wow, that sounds like a tricky bug indeed. Good to get that one fixed. Changing the test or the test output is fine either way, so this looks good.